clusterloader2/pkg/measurement/common/ooms_tracker.go

```diff
@@ -115,6 +123,56 @@ func (m *clusterOOMsTrackerMeasurement) String() string {
 	return clusterOOMsTrackerName
 }
 
+func (m *clusterOOMsTrackerMeasurement) getOOMsTrackerInformer(ctx context.Context, client clientset.Interface) cache.SharedInformer {
+	listFunc := func(options metav1.ListOptions) (runtime.Object, error) {
+		o := metav1.ListOptions{
+			Limit: 1,
+		}
+		result, err := client.CoreV1().Events(metav1.NamespaceAll).List(ctx, o)
+		if err != nil {
+			return nil, err
+		}
+		result.Continue = ""
+		result.Items = nil
+		return result, nil
+	}
+	watchFunc := func(options metav1.ListOptions) (watch.Interface, error) {
+		options.FieldSelector = m.selector.FieldSelector
+		return client.CoreV1().Events(metav1.NamespaceAll).Watch(ctx, options)
+	}
+	i := cache.NewSharedInformer(&cache.ListWatch{ListFunc: listFunc, WatchFunc: watchFunc}, nil, 0)
+	i.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			m.handleOOMEvent(obj)
+		},
+		UpdateFunc: func(_, obj interface{}) {
+			m.handleOOMEvent(obj)
+		},
+		DeleteFunc: func(_ interface{}) {},
+	})
+	return i
+}
+
+func (m *clusterOOMsTrackerMeasurement) handlePriorOOMs(ctx context.Context, client clientset.Interface) error {
+	pg := pager.New(pager.SimplePageFunc(func(opts metav1.ListOptions) (runtime.Object, error) {
+		return client.CoreV1().Events(metav1.NamespaceAll).List(ctx, opts)
+	}))
+	pg.PageSize = initialListPageSize
+
+	obj, _, err := pg.List(ctx, metav1.ListOptions{})
```

Comment:
Instead of Listing everything - you want to use EachListItem:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/tools/pager/pager.go#L156

Reply:
Done.